### PR TITLE
Live snapshot:Maximum snapshot chain check

### PR DESCRIPTION
**Pull request by aliang123**
blockdev_snapshot_chains:support maxium chain check

Signed-off-by: Aihua Liang <aliang@redhat.com>

id:1811519

### CODE REVIEW

File: qemu/tests/blockdev_snapshot_chains.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+from virttest import error_context
+
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+from provider.virt_storage.storage_admin import sp_admin
+
+
+class BlockdevSnapshotChainsTest(BlockDevSnapshotTest):
+
+    def __init__(self, test, params, env):
+        self.snapshot_num = int(params.get("snapshot_num"))
```
Comment (xutian) [inline qemu/tests/blockdev_snapshot_chains.py]: default value is required here,  convert None to int will raise exception.

File: qemu/tests/cfg/blockdev_snapshot_chains.cfg [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    only Linux
+    start_vm = yes
+    storage_type_default = "directory"
+    storage_pool = default
+    images += " data1"
+    image_size_data1 = 500M
+    image_name_data1 = "images/data1"
+    force_create_image_data1 = yes
+    force_remove_image_data1 = yes
+    rebase_mode = unsafe
```
Comment (xutian) [inline qemu/tests/cfg/blockdev_snapshot_chains.cfg]: is it really need this param ?

File: qemu/tests/blockdev_snapshot_chains.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            logging.error(str(error))
+
+
+def run(test, params, env):
+    """
+    Backup VM disk test when VM reboot
+
+    1) start VM with system disk
+    2) create target disk with qmp command
+    3) load stress in guest
```
Comment (xutian) [inline qemu/tests/blockdev_snapshot_chains.py]: Please update you steps in here.

File: qemu/tests/cfg/blockdev_snapshot_chains.cfg (hunk 2 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_snapshot_chains.cfg]: you mean "rebase_mode=unsafe"? Yes, need it, for there's a product bug if we use its default value.

File: qemu/tests/blockdev_snapshot_chains.py (hunk 3 above)
Comment (aliang123) [reply qemu/tests/blockdev_snapshot_chains.py]: Will update it later, thanks.

File: qemu/tests/blockdev_snapshot_chains.py (hunk 1 above)
Comment (aliang123) [reply qemu/tests/blockdev_snapshot_chains.py]: Will update it later, thanks.

File: qemu/tests/cfg/blockdev_snapshot_chains.cfg (hunk 2 above)
Comment (xutian) [reply qemu/tests/cfg/blockdev_snapshot_chains.cfg]: Got it.

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_chains.q35: PASS (162.42 s)

**Comment by aliang123** [comment]
@ngu-niny , @meet-lucky, please help to review, thanks.

**Comment by meet-lucky** [comment]
(1/1) Host_RHEL.m8.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_chains.q35: PASS (178.54 s)

Acked-by:  Qian Dong  qdong@redhat.com

**Comment by ngu-niny** [comment]
Test PASS:

 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.blockdev_snapshot_chains: PASS (313.25 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**